src/main/java/com/emc/pravega/perf/PerfStats.java

```diff
@@ -41,80 +39,132 @@
     private long bytes;
     private long maxLatency;
     private long totalLatency;
-    private long windowMaxLatency;
-    private long windowTotalLatency;
-    private long windowCount;
-    private long windowBytes;
-    final private long reportingInterval;
+    final private long windowInterval;
+    private timeWindow window;
+
+    /**
+     * private class for Performance statistics within a given time window.
+     */
+    private class timeWindow {
+        final private Instant startTime;
+        private Instant lastTime;
+        private long count;
+        private long bytes;
+        private long maxLatency;
+        private long totalLatency;
+
+        public timeWindow() {
+            this.startTime = Instant.now();
+            this.lastTime = this.startTime;
+            this.count = 0;
+            this.bytes = 0;
+            this.maxLatency = 0;
+            this.totalLatency = 0;
+        }
+
+        /**
+         * record the latency and bytes
+         *
+         * @param latency latency in ms.
+         * @param bytes   number of bytes.
+         */
+        public void record(long latency, long bytes) {
+            this.count++;
```

Comment:
In our discussions, we decided to have a raw file with the latency of operations. Maybe I'm missing something, but I'm not seeing the csv-like output of the operations. Maybe here, we need so write to a file all (or a evenly distributed fraction of) the operations:
```
Timestamp, Payload size, read/write, latency
1,100,0,20
2,100,0,23
...
``` 
Otherwise, we lose a lot of data to understand the behavior of latency (e.g., spike periodicity, configurable windows).

Reply:
yes, but I am thinking adding the metrics along with the csv reporter.. so, i think it should go as separate PR please.